src/modules/compare/gui/compareWidgetFuncs.cpp

```diff
@@ -0,0 +1,122 @@
+
+#include "gui/render/renderableData1D.h"
+#include "main/dissolve.h"
+#include "modules/compare/compare.h"
+#include "modules/compare/gui/compareWidget.h"
+#include <QDebug>
+
+CompareModuleWidget::CompareModuleWidget(QWidget *parent, CompareModule *module, Dissolve &dissolve)
+    : ModuleWidget(parent, dissolve), module_(module)
+{
+    // Set up user interface
+    ui_.setupUi(this);
+
+    refreshing_ = true;
+
+    // Set up button array
+    buttons_[0] = ui_.Data1DButton;
+    buttons_[1] = ui_.Data2DButton;
+    buttons_[2] = ui_.Data3DButton;
+
+    // Set up the main graph
+    graph_ = ui_.PlotWidget->dataViewer();
+
+    // -- Set view
+    graph_->view().setViewType(View::FlatXYView);
+    graph_->view().axes().setMax(0, 10.0);
+    graph_->view().axes().setMin(1, -1.0);
+    graph_->view().axes().setMax(1, 1.0);
+    graph_->view().setAutoFollowType(View::AllAutoFollow);
+    graph_->groupManager().setVerticalShiftAmount(RenderableGroupManager::TwoVerticalShift);
+
+    // -- Set group styling
+    graph_->groupManager().setGroupFixedColour("Delta", StockColours::BlackStockColour);
+    graph_->groupManager().setGroupColouring("Delta", RenderableGroup::FixedGroupColouring);
+    graph_->groupManager().setGroupStipple("Delta", LineStipple::QuarterDashStipple);
+    graph_->groupManager().setGroupVerticalShifting("Delta", RenderableGroup::IndividualVerticalShifting);
+
+    graph_->groupManager().setGroupFixedColour("DataA", StockColours::RedStockColour);
+    graph_->groupManager().setGroupColouring("DataA", RenderableGroup::FixedGroupColouring);
+    graph_->groupManager().setGroupVerticalShifting("DataA", RenderableGroup::IndividualVerticalShifting);
+
+    graph_->groupManager().setGroupFixedColour("DataB", StockColours::LightGreenStockColour);
+    graph_->groupManager().setGroupColouring("DataB", RenderableGroup::FixedGroupColouring);
+    graph_->groupManager().setGroupVerticalShifting("DataB", RenderableGroup::IndividualVerticalShifting);
+
+    refreshing_ = false;
+    updateControls(ModuleWidget::RecreateRenderablesFlag);
+}
+
+/*
+ * UI
+ */
+
+// Update controls within widget
+void CompareModuleWidget::updateControls(const Flags<ModuleWidget::UpdateFlags> &updateFlags)
+{
+
+    if (updateFlags.isSet(ModuleWidget::RecreateRenderablesFlag) || ui_.PlotWidget->dataViewer()->renderables().empty())
+    {
+        refreshing_ = true;
+
+        ui_.PlotWidget->clearRenderableData();
+
+        // Get the index of the checked button
+        auto index =
+            std::find_if(buttons_.begin(), buttons_.end(), [&](auto button) { return button->isChecked(); }) - buttons_.begin();
+
+        // If no buttons are checked, check the first one by default
+        if (index == buttons_.size() && buttons_.back()->isChecked() == false)
+        {
+            buttons_.front()->setChecked(true);
+            index = 0;
+        }
+
+        // Create the renderables for Data1D
+        if (index == 0)
```

Comment:
Why can't we just do:
```suggestion
        if (ui_.Data1DButton->isChecked())
```
You can set the state of the 1D button to be checked in Designer and presumably they're in a button group, so everything should work you want.